pkg/proxy/integrations/mongoparser/scramAuth.go

```diff
@@ -266,6 +292,11 @@ func handleSaslStart(i int, actualMsg map[string]interface{}, expectedRequestSec
 	logger.Debug("after replacing the new client nonce in auth response", zap.String("first response", newFirstAuthResponse))
 	// replace the payload with new first response auth
 	responseMsg["payload"].(map[string]interface{})["$binary"].(map[string]interface{})["base64"] = base64.StdEncoding.EncodeToString([]byte(newFirstAuthResponse))
+	responseMsg, err = updateConversationId(responseMsg, int(util.GetNextID()))
```

Comment:
For my clarity, the thing we should consider here mainly is that, the conversationId shouldn't be same. Every time it needs to change. Is this the only change done here ? and doesn't that depend on the client request message ..? Some more info regarding conversationId helps me understand  more regarding this change mainly like why it is used etc.

Reply:
Since scram is a challenge based authentication, so the challenge is done in conversation format. There are 2 request-response cycles: saslStart and saslContinue. During the response of saslStart the conversationId is passed by db server, and that is used by the saslContinue request. Using this conversationId the authMessage are mapped to generate the server proof. So this has to be unique so that we can map concurrent authMessage's. Earlier the field is not updated for the saslStart response.

Reply:
Got it.